### ISSUE TITLE
InvalidParameterValueException: We currently do not support adding policies for $LATEST.

### ISSUE THREAD

**Issue by agconti** [terraform-aws-modules/terraform-aws-lambda#36]
# The problem 
I'm running into an error while following trying to attach IAM policies to my lambda function. I've modeled my terraform off the complete example in the docs. The result is that they're really similar so I'm not sure how I'm running into this issue. I also see that you've handled this issue before this the [source](https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/master/main.tf#L175). 

## Error: 
```bash
Error: Error adding new Lambda Permission for sign-up: InvalidParameterValueException: We currently do not support adding policies for $LATEST.
{
  RespMetadata: {
    StatusCode: 400,
    RequestID: "ed82baaf-7fb8-4291-b2af-b8796d7bbea4"
  },
  Message_: "We currently do not support adding policies for $LATEST.",
  Type: "User"
}

  on .terraform/modules/lambda_function/terraform-aws-lambda-1.15.0/main.tf line 161, in resource "aws_lambda_permission" "current_version_triggers":
 161: resource "aws_lambda_permission" "current_version_triggers" {
```

## Terraform
```terraform
locals {
  table_name = "test"
}

module "lambda_function" {
  source = "terraform-aws-modules/lambda/aws"

  function_name = "sign-up"
  description   = "Allows users to sign up"
  handler       = "index.handler"
  runtime       = "nodejs12.x"
  publish       = true

  source_path = "../../index.js"
  environment_variables = {
    TABLE_NAME = local.table_name
  }

  allowed_triggers = {
    AllowExecutionFromAPIGateway = {
      service = "apigateway"
      arn     = aws_api_gateway_rest_api.sign_up.arn
    }
  }

  attach_policy_statements = true
  policy_statements = {
    dynamodb = {
      effect    = "Allow",
      actions   = ["dynamodb:PutItem"],
      resources = [module.dynamodb_table.this_dynamodb_table_arn]
    }
  }
}

resource "aws_api_gateway_rest_api" "sign_up" {
  name        = "Test rest API gateway"
  description   = "Test"

  endpoint_configuration {
    types = ["REGIONAL"]
  }
}

module "dynamodb_table" {
  source   = "terraform-aws-modules/dynamodb-table/aws"

  name     = local.table_name
  hash_key = "created"

  attributes = [
    {
      name = "created"
      type = "S"
    },
  ]
}
```

index.js:
```js
exports.handler =  async function(event, context) {
  console.log("EVENT: \n" + JSON.stringify(event, null, 2))
  return context.logStreamName
}
```
## Steps to reproduce

1. Copy terraform and index.js to testing directory
2. `terraform init`
3. `terraform plan`
4. See error. 

Please let me know if I can give you more information. Thanks for taking a look!

**Comment by antonbabenko** [comment]
Hi @agconti !

Thanks for opening this issue. I forgot to describe this case in the README, but I will do this soon.

Here is the solution:
When the Lambda function is created with `publish = true` the new version is automatically increased and a qualified identifier (version number) created and used when setting Lambda permissions.

When `publish = false` (default), only unqualified identifier (`$LATEST`) is available which leads to the error - `"We currently do not support adding policies for $LATEST."` 

The solution is to either disable the creation of allowed triggers for the current version:

```terraform
create_current_version_allowed_triggers = false
```

or to enable publish of Lambda function (`publish = true`).

**Comment by antonbabenko** [comment]
Fixed in dccacab

**Comment by agconti** [comment]
Hey @antonbabenko, this worked for me. Thanks for your help!
